src/hooks/staking/useVaults.tsx

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useMemo, useState } from 'react';
+import { Contract as MulticallContract, Provider as MulticallProvider } from '@pelith/ethers-multicall';
+import { useWeb3React } from '@web3-react/core';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import debounce from 'lodash.debounce';
+import { ChainDataFetchingState, NEW_SHAKKA_ADDRESSES, ChainId } from '../../constants';
+import STAKING_ABI from '../../constants/abis/shakka.json';
+import { useBlockNumber } from '../../state/application/hooks';
+
+export default function useVaults(): {
+  vaults: any;
+  vaultCount: any;
+  fetchDataState: ChainDataFetchingState;
+  } {
+    const { chainId } = useWeb3React();
+    const latestBlockNumber = useBlockNumber();
+    const [vaults, setVaults] = useState<any>();
+    const [vaultCount, setVaultCount] = useState<any>();
+    const [transactionSuccess, setTransactionSuccess] = useState(false);
+  
+    const fetchDataState: ChainDataFetchingState = useMemo(() => {
+      return transactionSuccess ? ChainDataFetchingState.SUCCESS : ChainDataFetchingState.LOADING;
+    }, [transactionSuccess]);
+
+    const providers = useMemo(() => {
+      const ethProvider = new JsonRpcProvider(process.env.REACT_APP_NETWORK_URL);
+      const bscProvider = new JsonRpcProvider(process.env.REACT_APP_BSC_NETWORK_URL);
+      const polygonProvider = new JsonRpcProvider(process.env.REACT_APP_POLYGON_NETWORK_URL);
+      const kovanProvider = new JsonRpcProvider(process.env.REACT_APP_KOVAN_NETWORK_URL);
+      return {[ChainId.MAINNET]: ethProvider, [ChainId.BSC]: bscProvider, [ChainId.POLYGON]: polygonProvider, [ChainId.KOVAN]: kovanProvider};
+    }, [])
+
+    const fetchVaults = async () => {
```

Comment:
should set chainId to the function
```
const fetchVaults = async (chainId: ChainId) => {}
```

then wrap this by using `useCallback`, then line: 48 can remove fetchVaults from the dependencies.
afterall, line 50 `useEffect` should update with new dependencies.

this way can make users more clarify which is a constants value or `react state`. and avoid `react closure` issue